src/PKSim.Core/Snapshots/Services/SnapshotSerializer.cs

```diff
@@ -1,12 +1,16 @@
-﻿using System.IO;
+﻿using System;
+using System.IO;
+using System.Linq;
 using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Schema;
 
 namespace PKSim.Core.Snapshots.Services
 {
    public interface ISnapshotSerializer
    {
       void Serialize(object snapshot, string fileName);
-      T Deserialize<T>(string fileName);
+      object[] DeserializeAsArray(string fileName, Type snapshotType);
```

Comment:
I am just implementing what I need now. I want to ensure that we can deserialize both a single item and an array of items.